### PR TITLE
Improved error handling

### PR DESCRIPTION
**Pull request by julesce**
**Why?** There is a fair bit of duplication going on around how errors are being handling in the various stores. This PR aims to fix that, make things more consistent, and handle `Faraday::ParsingError, Faraday::SSLError` errors properly as well.

### CODE REVIEW

**Comment by julesce** [comment]
@scorps69 @marcheiligers @sruli @maximgladkov - How do you guys feel about these changes?

**Comment by sruli** [comment]
👀

**Comment by sruli** [comment]
@julesce in terms of the tests, is the idea to include those in this PR? I feel like we should probably add them here while we're add it since this code doesn't get updated too often and I suspect that if we don't do it now, well...

File: lib/mad_cart/store/base.rb [hunk 1]
```diff
@@ -42,6 +42,41 @@ def init_connection
         @connection ||= execute_delegate(klass.connection_delegate, @init_args)
       end
 
+      def validate_by_path(path)
+        check_for_errors do
+          connection.get(path)
+        end
+        return true
```
Comment (sruli) [inline lib/mad_cart/store/base.rb]: Any reason why we're explicitly returning true here? Also in the rescue block?

**Comment by julesce** [comment]
I should clarify that I'll be hooking up the tests if you guys are happy with the underlying refactor 😸

File: lib/mad_cart/store/base.rb (hunk 1 above)
Comment (julesce) [inline lib/mad_cart/store/base.rb]: This method gets called by the `valid?` method in the store implementation. Perhaps renaming this method to `valid_by_path?` would make more sense?

Comment (sruli) [inline lib/mad_cart/store/base.rb]: I was specifically referring to the use of the `return` keyword. Returning the values `true` and `false` is cool, but do we need `return`?

Comment (julesce) [inline lib/mad_cart/store/base.rb]: Ah yeah sure, totally nuking the `return` :-)

**Comment by sruli** [comment]
Looks cool to me 👍

**Comment by julesce** [comment]
I've taken this for a spin on `gdtest` and all looks good. Merging.

**Pull request merged**